### DISCUSSION TITLE
Integrating Leafmap into a Shiny for Python Application

### DISCUSSION THREAD

**Discussion by diehl** [opengeos/leafmap#942] [Q&A] [answered]
Is it possible to integrate Leafmap into a Shiny for Python application, particularly one supporting interactive exploration? I would want to get click events sent back to the application at a minimum. Being able to annotate the map would be next level.

**Comment by giswqs** [comment]
See this: https://github.com/opengeos/leafmap/issues/875

**Comment by giswqs** [comment, marked as answer]
Here is a shiny-leafmap example: https://huggingface.co/spaces/giswqs/shiny-leafmap

**Reply by diehl** [reply]
Thank you @giswqs for the quick reply!

**Reply by diehl** [reply]
@giswqs Must one use the ipyleaflet backend to integrate Leafmap with Shiny for Python?

**Reply by giswqs** [reply]
I am not an active Shiny user. Not sure about this. If Shiny supports HTML, I guess it should work fine for most leafmap mapping backends.

**Reply by diehl** [reply]
@giswqs Roger that. I noticed the Python bindings for MapLibre support Shiny integration. https://eoda-dev.github.io/py-maplibregl/shiny/ So I thought it might be possible for Leafmap as well.
